app/src/main/java/net/aiscope/gdd_app/ui/metadata/SampleImagesAdapter.kt

```diff
@@ -80,6 +81,7 @@ private class ImageViewHolder(
                 itemView.context.resources.getDimensionPixelSize(R.dimen.sample_image_thumbnail_height)
             )
             itemView.setImageBitmap(bitmap)
+            itemView.setOnClickListener { onImageClicked(index) }
```

Comment:
@eduardb So this now uses the actual files rather than the index. I still have an `onClickListener` for each bind though. Not entirely sure how to improve this.

Reply:
OK updated. See if this is what you meant